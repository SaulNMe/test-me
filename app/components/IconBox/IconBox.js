import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { LinearGradient } from 'expo-linear-gradient';

import styles from './IconBoxStyle';
import { Feather } from '@expo/vector-icons';
import { Colors } from 'qualificame-native/app/styles';

export default class IconBox extends PureComponent {
	render() {
		let gradientColors;
		switch(this.props.bgColor) {
			case 'green':
				gradientColors = ['#0F9F92', '#12B184'];
			break;
			case 'blue':
				gradientColors = ['#154BBA', '#107EC2'];
			break;
			case 'orange':
				gradientColors = ['#E62808', '#FC680A'];
			break;
			case 'pink':
				gradientColors = ['#E31862', '#EB194D'];
			break;
			case 'gray':
				gradientColors = ['#ccc', '#ccc'];
			break;
			default:
				gradientColors = ['#fff', '#fff'];
			break;
		}
		return (
			<LinearGradient colors={gradientColors} style={[styles.defaultSize, styles.centerObjects]} start={[0, 0]} end={[1, 0]}>
				<Feather name={this.props.iconName} size={24} color={Colors.white}/>
			</LinearGradient>
		);
	}
}

	IconBox.propTypes = {
		bgColor: PropTypes.string,
		iconName: PropTypes.string

	}

	IconBox.defaultProps = {
		bgColor: 'green',
		iconName: 'x'
	}
