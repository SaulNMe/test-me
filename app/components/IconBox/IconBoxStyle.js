import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.flexBox,
	defaultSize:{
		width: 32,
		height: 32,
		borderRadius: 5
	}
});
