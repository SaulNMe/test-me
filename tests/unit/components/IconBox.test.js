import React from 'react';
import { shallow } from 'enzyme';
import { LinearGradient } from 'expo-linear-gradient';

import IconBox from 'qualificame-native/app/components/IconBox';
import { Feather } from '@expo/vector-icons';

function createTestProps (props) {
	return {
		// common props
		bgColor: 'blue',
		iconName: 'moon',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<IconBox {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays the correct feather icon ', () => {
		expect(wrapper.find(Feather).exists()).toBeTruthy();
		expect(wrapper.find(Feather).props().name).toEqual('moon')
	});

	it('displays the correct background color', () => {
		expect(wrapper.find(LinearGradient).props().colors).toEqual([ '#154BBA', '#107EC2' ]);
	});	

	describe('when props are not set', () => {
		beforeEach(() => {
			wrapper = createWrapper();
		});
	
		it('displays the correct feather icon', () => {
			expect(wrapper.find(Feather).exists()).toBeTruthy();
			expect(wrapper.find(Feather).props().name).toEqual('x')
		});	

		it('displays the correct background color', () => {
			expect(wrapper.find(LinearGradient).props().colors).toEqual([ '#0F9F92', '#12B184' ]);
		});	

	});
})
