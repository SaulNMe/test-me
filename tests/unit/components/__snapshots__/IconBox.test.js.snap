// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<LinearGradient
  colors={
    Array [
      "#154BBA",
      "#107EC2",
    ]
  }
  end={
    Array [
      1,
      0,
    ]
  }
  start={
    Array [
      0,
      0,
    ]
  }
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 32,
        "width": 32,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Icon
    allowFontScaling={false}
    color="#fff"
    name="moon"
    size={24}
  />
</LinearGradient>
`;
